packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -568,11 +580,81 @@ class PlanarFreehandROITool extends AnnotationTool {
     const isEditingClosed = this.isEditingClosed;
 
     if (!(isDrawing || isEditingOpen || isEditingClosed)) {
+      const isPreScaled = isViewportPreScaled(viewport, targetId);
+
       // No annotations are currently being modified, so we can just use the
       // render contour method to render all of them
-      annotations.forEach((annotation) =>
-        this.renderContour(enabledElement, svgDrawingHelper, annotation)
-      );
+      annotations.forEach((annotation) => {
+        this.renderContour(enabledElement, svgDrawingHelper, annotation);
+
+        if (!this.configuration.calculateStats) return;
+
+        const { data } = annotation;
```

Comment:
Could you also please move all the new code below into a new function (eg: `renderStats` or something like that) leaving only two functions calls in the `forEach` loop. That helps having smaller and more specific functions that are easier to maintain.

Reply:
This is resolved, @lscoder